File: src/scala/src/main/scala/simplifying/conditional/expressions/PriceCalculator.scala
package simplifying.conditional.expressions

object PriceCalculator {

  def calculatePrice(price: Double, isSaleDay: Boolean): Double = {
    var total = .0
    if (isSaleDay) {
      total = price * 0.5
      notifySales(total)
    } else {
      total = price * 0.98
      notifySales(total)
    }
    total
  }

  private def notifySales(amount: Double): Unit = {
    println(s"Notify sales $amount")
  }
}

File: src/java/main/java/simplifying/conditional/expressions/PriceCalculator.java
package simplifying.conditional.expressions;

import lombok.AllArgsConstructor;

@AllArgsConstructor
public class PriceCalculator {
    private static final double SALE_DISCOUNT = 0.5;
    private static final double NORMAL_DAY_DISCOUNT = 0.98;
    private final boolean isSaleDay;

    public double calculatePrice(double price) {
        double total = price * (isSaleDay ? SALE_DISCOUNT : NORMAL_DAY_DISCOUNT);
        notifySales(total);
        
        return total;
    }

    private void notifySales(double amount) {

    }
}
